gate/internal/use_case/system_token/revoke_system_token.go

```
@@ -0,0 +1,46 @@
+package system_token
+
+import (
+	"time"
+
+	"github.com/ZupIT/charlescd/gate/internal/logging"
+	"github.com/ZupIT/charlescd/gate/internal/repository"
+	"github.com/google/uuid"
+)
+
+type RevokeSystemToken interface {
+	Execute(id uuid.UUID) error
+}
+
+type revokeSystemToken struct {
+	systemTokenRepository repository.SystemTokenRepository
+}
+
+func NewRevokeSystemToken(repository repository.SystemTokenRepository) revokeSystemToken {
+	return revokeSystemToken{
+		systemTokenRepository: repository,
+	}
+}
+
+func (r revokeSystemToken) Execute(id uuid.UUID) error {
+	systemToken, err := r.systemTokenRepository.FindById(id)
+
+	if err != nil {
+		return logging.WithOperation(err, "RevokeSystemToken.Execute")
+	}
+
+	if !systemToken.Revoked {
```

we could return an error if this condition isn't true, what do you think?
Ex. "Token is alrealdy inactive" or "Token has already been revoked"

---

A 422 code?

---

I agree with Pedro, we could return a 422 in this scenario

---

Agreed